### PR TITLE
Astro 2972 segmented button audit

### PR DESCRIPTION
**Pull request by micahjones13**
## Brief Description

Adds missing styles to rux-segmented button found during the audit. 
- Adds a size prop similar to rux-button's
    - The default size styling is small since that's how it was styled before this change. This will be updated to default to 
       medium in 7.0
- Updates the heights for each size to match figma 
- Adds hover state overrides to all other states, as per figma (that was a doozy of some css lemme tell ya)
- Updates the rux-segmented button story to include medium and large variants 
- Updates docs on segmented button story/readme to warn against supplying identical labels in the data prop

I also ran a stencil build, so there are some random changed files in this.
## JIRA Link

https://rocketcom.atlassian.net/browse/ASTRO-2974

## General Notes

CSS is a bit convoluted now, I left comments to explain what I was doing. Let me know if you want those comments removed

## Motivation and Context

dev -> design audit 

## Types of changes

- [ ] Bug fix
- [x] New feature
- [ ] Breaking change

## Checklist

- [x] This PR adds or removes a Storybook story.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: packages/web-components/src/stories/segmented-button.stories.mdx [hunk 1]
```diff
@@ -46,20 +47,95 @@ export const SegmentedButton = (args) => {
...
+        name="Large"
+    >
+        {Large.bind()}
+    </Story>
+</Canvas>
+
 ## Usage
 
-Pass an array of segments via the `data` attribute on the Segmented Button custom element. Segment items in the `data` array must be objects with a `label` string. The first item in the array will be auto-selected unless another segment item has a `selected` property with a truthy value
+Pass an array of segments via the `data` attribute on the Segmented Button custom element. Segment items in the `data` array must be objects with a `label` string. The first item in the array will be auto-selected unless another segment item has a `selected` property with a truthy value. Avoid labels with the same text, as this will interfere with selecting segments.
```
Comment (markacianfrani) [inline packages/web-components/src/stories/segmented-button.stories.mdx:138]: good catch

File: packages/web-components/src/components/rux-segmented-button/rux-segmented-button.tsx [hunk 2]
```diff
@@ -30,6 +30,13 @@ export class RuxSegmentedButton {
      */
     @Prop({ reflect: true, mutable: true }) selected: string = ''
 
+    /**
+     * Changes size of segmented button from medium to small or large by setting sizing classes
```
Comment (markacianfrani) [inline packages/web-components/src/components/rux-segmented-button/rux-segmented-button.tsx:34]: nit: I would just say changes the size here. the specific classes are just implementation details and hidden by the shadow dom

File: packages/web-components/src/components/rux-segmented-button/rux-segmented-button.scss [hunk 3]
```diff
@@ -51,21 +51,42 @@
...
+    // Default is small for 6.0 - will update default to be medium in 7.0
+    height: 26px;
     overflow: hidden;
     padding: 0;
     margin: 0;
     list-style: none;
     border-radius: var(--segmented-button-border-radius);
     background-color: var(--segmented-button-background-color);
+    &--large {
+        height: 46px;
```
Comment (markacianfrani) [inline packages/web-components/src/components/rux-segmented-button/rux-segmented-button.scss:63]: can we replace all these px w/ rem?

File: packages/web-components/src/components/rux-segmented-button/rux-segmented-button.tsx (hunk 2 above)
Comment (micahjones13) [reply packages/web-components/src/components/rux-segmented-button/rux-segmented-button.tsx:34]: I'm fine with that. I based it off of the rux-button size description. Want me to change that too?

File: packages/web-components/src/components/rux-segmented-button/rux-segmented-button.scss (hunk 3 above)
Comment (micahjones13) [reply packages/web-components/src/components/rux-segmented-button/rux-segmented-button.scss:63]: Yes my b 😞

**Review by markacianfrani — APPROVED** [review]

**Pull request merged**